pkg/build/builder/sti.go

```diff
@@ -378,6 +378,13 @@ func s2iBuildLabels(build *buildapiv1.Build, sourceInfo *git.SourceInfo) map[str
 	}
 
 	labels = s2iutil.GenerateLabelsFromSourceInfo(labels, toS2ISourceInfo(sourceInfo), builderutil.DefaultDockerLabelNamespace)
+	if build != nil && build.Spec.Source.Git != nil && build.Spec.Source.Git.Ref != "" {
+		// override the io.openshift.build.commit.ref label to match what we
+		// were originally told to check out, as well as the
+		// OPENSHIFT_BUILD_REFERENCE environment variable.  This can sometimes
+		// differ from git's view (see PotentialPRRetryAsFetch for details).
+		labels[builderutil.DefaultDockerLabelNamespace+"build.commit.ref"] = build.Spec.Source.Git.Ref
```

Comment:
i'm guessing docker.go also has similar logic that needs to be updated?

Reply:
updated, thanks